tests/typecheck/models/test_contrib_models.yml

```diff
@@ -31,9 +31,11 @@
 
 -   case: can_override_abstract_user_manager
     main: |
-        from myapp.models import User
-        reveal_type(User.objects)  # N: Revealed type is "myapp.models.UserManager[myapp.models.User]"
-        reveal_type(User.objects.all())  # N: Revealed type is "django.db.models.query._QuerySet[myapp.models.User, myapp.models.User]"
+        from myapp.models import MyBaseUser, MyUser
+        reveal_type(MyBaseUser.objects)  # N: Revealed type is "myapp.models.MyBaseUserManager[myapp.models.MyBaseUser]"
```

Comment:
Similarly to my [note](https://github.com/typeddjango/django-stubs/pull/1649#issuecomment-1681846078) on the other PR, this is a bogus type, since `MyBaseUserManager` is not a generic class and does not accept type parameters. There are many existing instances of this problem, so maybe we don’t care about adding new ones to the test suite, but I figure it’s worth pointing out.

Reply:
I thought so too, but I think mypy is happy about that: https://github.com/typeddjango/django-stubs/pull/1666#discussion_r1311390430

Here it is under a playground link.
https://mypy-play.net/?mypy=latest&python=3.11&gist=1fc5abb96d6e8c606849f3f1af181196

Reply:
Everything you wrote in that playground is fine. `A` is generic (explicitly), `B` is generic (because it inherits from `A` and does not bind the free type parameter `T`), `C` is not generic (because it inherits from `B` but **does** bind the free type parameter `T` to `int`).

What would not be fine is `class D(C[int]): ...`; that would be similar to the problem I’m pointing out here.

Reply:
Hm, or maybe that's off. Since the type is already declared for the argument i.e. `MyUser`.

This should really only be `MyBaseUserManager`. The plugin should refrain from add in model arguments when there's no `TypeVar`

Reply:
Anyways, I wanted to add these reveals to follow what happened with managers on `AbstractUser`.

I'll try to have a follow up round to fix sprinkling of model arguments.

Reply:
I have opened a draft for this just now: #1683